CredentialsManager/Manager.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using CredentialsManager.FilesExceptions;
+using CredentialsManager.Models;
+
+namespace CredentialsManager
+{
+  public static class Manager
+  {
+    private const string FieldDelimiter = ",";
+    private const string CredentialsFile = "Users.txt";
+    private static List<Credentials> _credentials = new();
+
+    public static void Initialize()
+    {
+      _credentials = GetCredentials();
+      _credentials.Sort();
+  
+      CheckForDuplicates();
+    }
+
+    private static List<Credentials> GetCredentials()
+    {
+      var content = Files.ReadAllLines(CredentialsFile);
+      var credentials = new List<Credentials>();
+
+      foreach (var row in content)
+      {
+        var isValidCredential = Credentials.TryParse(row, FieldDelimiter, out var credential);
+        if (isValidCredential)
+        {
+          credentials.Add(credential);
+        }
+      }
+
+      return credentials;
+    }
+
+    private static void CheckForDuplicates()
+    {
+      for (var index = 0; index + 1 < _credentials.Count; index++)
+      {
+        var currentCredential = _credentials[index];
+        var nextCredential = _credentials[index + 1];
+
+        if (currentCredential.UserName.Equals(nextCredential.UserName))
+        {
+          throw new DuplicateUserCredentialsException("Duplicate user names found!");
+        }
+      }
+      
+    }
+
+    public static bool Login(Credentials credentials)
+    {
+      return _credentials.Contains(credentials);
+    }
+
+    public static void Register(Credentials credentials)
+    {
+      if (IsUserNameTaken(credentials))
+      {
+        throw new UserNameIsTakenException("User name is taken!");
+      }
+
+      Files.WriteLine(CredentialsFile, CreateString(credentials), true);
+    }
+
+    private static bool IsUserNameTaken(Credentials credentials)
+    {
+      foreach (var storedCredential in _credentials)
+      {
+        if (storedCredential.UserName.Equals(credentials.UserName))
```

Comment:
You might want to do a case insensitive comparison here.

Reply:
I lost this one in translation.
Before I refactored with OOP I had this  method
```
private static bool IsUserNameEqual(string[] left, string[] right)
{
  return left[0].Equals(right[0], StringComparison.OrdinalIgnoreCase);
}
```
But now that I implemented `IComparable` I can use `currentCredential.CompareTo(nextCredential).Equals(0)`

Reply:
I updated it in commit [712418a](https://github.com/csinn/CSharp-From-Zero-To-Hero-v2/pull/43/commits/712418a54e8d87bd0bfb26656174ddacc93c2d3f)